### PR TITLE
[SPARK-6428] Turn on explicit type checking for public methods.

### PR DESCRIPTION
**Pull request by rxin**
This builds on my earlier pull requests and turns on the explicit type checking in scalastyle.

### CODE REVIEW

File: graphx/src/main/scala/org/apache/spark/graphx/lib/ConnectedComponents.scala [hunk 1]
```diff
@@ -37,7 +37,7 @@ object ConnectedComponents {
    */
   def run[VD: ClassTag, ED: ClassTag](graph: Graph[VD, ED]): Graph[VertexId, ED] = {
     val ccGraph = graph.mapVertices { case (vid, _) => vid }
-    def sendMessage(edge: EdgeTriplet[VertexId, ED]) = {
+    def sendMessage(edge: EdgeTriplet[VertexId, ED]): Iterator[(VertexId, VertexId)] = {
```
Comment (ankurdave) [inline graphx/src/main/scala/org/apache/spark/graphx/lib/ConnectedComponents.scala]: This change is fine, but hopefully the rule doesn't require it -- `sendMessage` isn't a public method here.

**Comment by ankurdave** [comment]
LGTM for the GraphX changes.

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala [hunk 2]
```diff
@@ -59,7 +59,7 @@ class ExecutorRunnable(
   val yarnConf: YarnConfiguration = new YarnConfiguration(conf)
   lazy val env = prepareEnvironment(container)
 
-  def run = {
+  def run(): Unit = {
```
Comment (mengxr) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala]: Is it an incompatible change?

File: core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 package org.apache.spark.api.java
 
+import java.lang
```
Comment (mengxr) [inline core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala]: minor: `java.{lang => jl}`? Just want to be consistent in the codebase.

**Comment by mengxr** [comment]
LGTM

**Pull request closed without merging**